src/components/Checkbox/Checkbox.tsx

```diff
@@ -1,117 +1,227 @@
 import clsx from 'clsx';
-import React, {
-  useRef,
-  useEffect,
-  ChangeEventHandler,
-  MutableRefObject,
-} from 'react';
+import React from 'react';
+import { useUID } from 'react-uid';
 import styles from './Checkbox.module.css';
-import { ENTER_KEYCODE } from '../../util/keycodes';
 
-export interface Props {
-  /**
-   * HTML id of the helper text used to describe the component
-   */
-  ariaDescribedBy?: string;
+/**
+ * Allow all native HTML input attributes,
+ * except for ones where we override type or strictness
+ */
+type CheckboxHTMLElementProps = Omit<
+  React.InputHTMLAttributes<HTMLInputElement>,
+  'checked' | 'id' | 'size'
+>;
+
+type InputProps = CheckboxHTMLElementProps & {
   /**
-   * Toggles whether or not the checkbox is checked or unchecked
+   * Whether checkbox is checked. Defaults to false.
+   * "indeterminate" can be used when a checkbox visually represents
+   * a list of checkboxes that is "partially" checked.
    */
-  checked?: boolean;
+  checked?: boolean | 'indeterminate';
   /**
-   * CSS class names that can be appended to the component.
+   * Additional classnames passed in for styling.
    */
   className?: string;
   /**
-   * Disables the field and prevents editing the contents
+   * Checkbox ID. Used to connect the input with a label for accessibility purposes.
    */
-  disabled?: boolean;
+  id: string;
+};
+
+type LabelProps = {
   /**
-   * HTML id for the component
+   * Text to render in label.
    */
-  id?: string;
+  text: React.ReactNode;
   /**
-   * Indeterminate prop that turns the checkbox indeterminate state on
+   * Additional classnames passed in for styling.
    */
-  indeterminate?: boolean;
```

Comment:
BM used a prop for indeterminate state but v0 uses checked prop = `'indeterminate'`, stuck with v0 style since we don't seem to have any extra styling for it besides an svg change for the checkmark